docs/design.rst

```diff
@@ -14,40 +14,46 @@ Class diagram
 
     @startuml
     class WarningsPlugin {
-        #checkers : WarningsChecker
-        +main()
+        #checkerList : WarningsChecker
     }
 
     class WarningsChecker {
         #min_count = 0
         #max_count = 0
         #count = 0
 
-        #{abstract} __init__(regex=None)
+        #{abstract} __init__(name)
         +set_limits(min_count=0, max_count=0,
-        +check(line)
+        +check(content)
         +get_count()
     }
 
-    class SphinxWarningsChecker {
+    class RegexChecker {
+        #{abstract} __init__(name, regex)
```

Comment:
It is abstract in the WarningsChecker class, and actually implemented in the RegexChecker class. So, I think it adds a lot of value to add it here as well.